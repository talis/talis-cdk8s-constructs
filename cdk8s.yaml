language: typescript
imports:
  - k8s@1.30.0
