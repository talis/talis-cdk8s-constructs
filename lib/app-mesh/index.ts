export * from "./ingress-gateway-nlb";
export * from "./app-service";
