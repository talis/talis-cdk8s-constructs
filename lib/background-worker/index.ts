export * from "./background-worker-props";
export * from "./background-worker";
