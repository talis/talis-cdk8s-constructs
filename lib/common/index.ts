export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
