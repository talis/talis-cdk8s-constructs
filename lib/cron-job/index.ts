export * from "./cron-job-props";
export * from "./cron-job";
