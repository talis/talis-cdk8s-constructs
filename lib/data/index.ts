export * from "./config-map";
export * from "./secret";
export * from "./factory";
