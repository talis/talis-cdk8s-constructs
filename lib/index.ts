export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
