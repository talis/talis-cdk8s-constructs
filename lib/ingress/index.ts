export * from "./ingress-props";
export * from "./ingress";
