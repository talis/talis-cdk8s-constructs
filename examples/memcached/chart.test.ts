import { MemcachedChart } from "./chart";
import { Testing } from "cdk8s";
import { TalisShortRegion, TalisDeploymentEnvironment } from "../../lib";

describe("Memcached example", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("Snapshot", () => {
    const app = Testing.app();
    const chart = new MemcachedChart(app, {
      environment: TalisDeploymentEnvironment.DEVELOPMENT,
      region: TalisShortRegion.LOCAL,
      watermark: "test",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
