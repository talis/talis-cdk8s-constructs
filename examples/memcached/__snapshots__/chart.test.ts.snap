// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Memcached example > Snapshot 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": {
      "labels": {
        "app": "example-memcached-app",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "test",
        "service": "example-memcached-app-development-local",
      },
      "name": "example-memcached-app-test",
      "namespace": "example-memcached-app-test",
    },
  },
  {
    "apiVersion": "v1",
    "kind": "ResourceQuota",
    "metadata": {
      "labels": {
        "app": "example-memcached-app",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "example-memcached-app-development-local",
      },
      "name": "quota",
      "namespace": "example-memcached-app-test",
    },
    "spec": {
      "hard": {
        "cpu": "100m",
        "memory": "128Mi",
        "pods": 2,
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "app": "example-memcached-app",
        "environment": "development",
        "instance": "memcached-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "1.5.20",
        "role": "memcached",
        "service": "example-memcached-app-development-local",
      },
      "name": "memcached-example",
      "namespace": "example-memcached-app-test",
    },
    "spec": {
      "clusterIP": "None",
      "ports": [
        {
          "port": 11211,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "app": "example-memcached-app",
        "instance": "memcached-example",
        "role": "memcached",
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": {
      "labels": {
        "app": "example-memcached-app",
        "environment": "development",
        "instance": "memcached-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "1.5.20",
        "role": "memcached",
        "service": "example-memcached-app-development-local",
      },
      "name": "memcached-example-sts",
      "namespace": "example-memcached-app-test",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "app": "example-memcached-app",
          "instance": "memcached-example",
          "role": "memcached",
        },
      },
      "serviceName": "memcached-example",
      "template": {
        "metadata": {
          "labels": {
            "app": "example-memcached-app",
            "environment": "development",
            "instance": "memcached-example",
            "managed-by": "cdk8s",
            "region": "local",
            "release": "1.5.20",
            "role": "memcached",
            "service": "example-memcached-app-development-local",
          },
        },
        "spec": {
          "containers": [
            {
              "args": [
                "--memory-limit=64",
              ],
              "image": "memcached:1.5.20",
              "livenessProbe": {
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "tcpSocket": {
                  "port": 11211,
                },
                "timeoutSeconds": 5,
              },
              "name": "memcached",
              "ports": [
                {
                  "containerPort": 11211,
                },
              ],
              "readinessProbe": {
                "failureThreshold": 5,
                "initialDelaySeconds": 5,
                "tcpSocket": {
                  "port": 11211,
                },
                "timeoutSeconds": 5,
              },
              "resources": {
                "limits": {
                  "cpu": "50m",
                  "memory": "64Mi",
                },
              },
            },
          ],
          "priorityClassName": "database",
        },
      },
    },
  },
]
`;
